### PR TITLE
Feat: Delete Organisation

### PR DESCRIPTION
**Pull request by IgweWisdomJaminel**
## How should this be manually tested
- Ensure you have the latest code from the `main` branch.
- Ensure your environment variables for database and authentication are set correctly.

### Steps to Test
1. **Clone the repository**: `git clone <repository-url>`
2. **Checkout to the new branch**: `git checkout <your-branch-name>`
3. **Compile the application 'javac application.java' 
4. **Run the application**: `java application.java start`
5. **Authenticate and obtain a valid token**: Use the login endpoint to get a token.
6. **Issue a DELETE request** to the endpoint:
    - Use a tool like Postman or curl:
      ```
      DELETE /api/v1/organizations/<org_id>
      Content-Type: application/json
      Authorization: Bearer <token>
      ```
    - Replace `<org_id>` with a valid organization ID and `<token>` with your authentication token.
7. **Verify the responses**:
    - For a valid request, expect a **204 No Content** status.
    - For an invalid organization ID format, expect a **400 Bad Request** status.
    - For a non-existent organization ID, expect a **404 Not Found** status.
    - For unauthorized deletion attempt, expect a **401 Unauthorized** status.

### Expected Outcomes
- **204 No Content**: The organization is marked as deleted.
- **400 Bad Request**: Invalid organization ID format.
- **404 Not Found**: Invalid organization ID.
- **401 Unauthorized**: User not authorized to delete this organization.

## Checklist

- [x] Developed server-side logic to handle organization deletion requests.
- [x] Validated and sanitized incoming organization ID data.
- [x] Validated that the user deleting the organization is the owner.
- [x] Changed the specified organization's `deleted` attribute to `true` in the database upon successful validation.
- [x] Maintained data integrity by keeping all related data of the specified organization intact.
- [x] Wrote unit tests to ensure the organization deletion endpoint validates input correctly and removes the organization from the system.
- [x] Tested various scenarios for submitting the organization ID (e.g., valid ID, non-existent ID, malformed ID, etc.).
- [x] Verified that all unit tests pass.
- [x] Ensured the code follows the project's style guidelines.
- [x] Removed all console logs and comments.
- [x] Updated the documentation as necessary.
- [x] Informed relevant stakeholders of the changes.

## Reference issue
[https://github.com/hngprojects/hng_boilerplate_java_web/issues/18
](url)

### CODE REVIEW

**Comment by lurldgbodex** [comment]
This is a java repo. Why do you have npm install?

**Comment by IgweWisdomJaminel** [comment]
Changed 🫣

**Comment by lurldgbodex** [comment]
please resolve your conflicts

**Comment by IgweWisdomJaminel** [comment]
Done My boss The conflict was application.properties someone deleted it and I think there was an update to the dockerfile

**Comment by lurldgbodex** [comment]
LGTM

**Review by Am0du — CHANGES_REQUESTED** [review]
Implement that change and we're good

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,33 @@
+spring.application.name=hng-java-boilerplate
+
```
Comment (lurldgbodex) [inline src/main/resources/application.properties:2]: remove the application.properties file

**Pull request closed without merging**